apps/aetx/src/aetx.erl

```diff
@@ -168,11 +175,21 @@ accounts(#aetx{ cb = CB, tx = Tx }) ->
 signers(#aetx{ cb = CB, tx = Tx }, Trees) ->
     CB:signers(Tx, Trees).
 
+%% We let 0 represent no TTL/infinity in order to keep the serialization as
+%% short as possible. Since there are no transactions in the genesis block
+%% there is little risk for confusion.
+-spec ttl(Tx :: tx()) -> max_ttl | aec_blocks:height().
+ttl(#aetx{ cb = CB, tx = Tx }) ->
+    case CB:ttl(Tx) of
+        0 -> max_ttl;
+        N -> N
+    end.
+
 -spec check(Tx :: tx(), Trees :: aec_trees:trees(), Height :: non_neg_integer(),
             ConsensusVersion :: non_neg_integer()) ->
     {ok, NewTrees :: aec_trees:trees()} | {error, Reason :: term()}.
-check(#aetx{ cb = CB, tx = Tx }, Trees, Height, ConsensusVersion) ->
-    case {CB:fee(Tx) >= aec_governance:minimum_tx_fee(), CB:ttl(Tx) >= Height} of
+check(AeTx = #aetx{ cb = CB, tx = Tx }, Trees, Height, ConsensusVersion) ->
+    case {CB:fee(Tx) >= aec_governance:minimum_tx_fee(), ttl(AeTx) >= Height} of
```

Comment:
`max_ttl > 1234` ?

Reply:
```
1> max_ttl > 1243.
true
```

Reply:
From http://erlang.org/doc/reference_manual/expressions.html#id81948
> number < atom < reference < fun < port < pid < tuple < map < nil < list < bit string

So `max_ttl` will always be greater than a block height. Though may impact readability.